File: arshGoyalDsaSheet/LongestCommonPrefix.py
from ast import List
class Solution:
   def longestCommonPrefix(self, v: List[str]) -> str:
       ans=""
       v=sorted(v)
       first=v[0]
       last=v[-1]
       for i in range(min(len(first),len(last))):
           if(first[i]!=last[i]):
               return ans
           ans+=first[i]
       return ans 

# Get input from the user
num_strings = int(input("Enter the number of strings: "))
v = []
for i in range(num_strings):
   string = input("Enter string {}: ".format(i + 1))
   v.append(string)

# Create an instance of the Solution class
solution = Solution()

# Find the longest common prefix
result = solution.longestCommonPrefix(v)

# Print the result
print("The longest common prefix is:", result)

File: NeetCodeALL/ArrayAndHashing/LongestCommonPrefix.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans="";
        sort(strs.begin(),strs.end());
        int n=strs.size();
        string first=strs[0],last=strs[n-1];
        for(int i=0;i<min(first.size(),last.size());i++){
            if(first[i]!=last[i]){
                return ans;
            }
            ans+=first[i];
        }
        return ans;
    }
};
int main(){
    int n;
    cin>>n;

    vector<string>strs(n);
    for(int i = 0;i<n;i++){
        cin>>strs[i];
    }

    Solution solution;
    string result = solution.longestCommonPrefix(strs);
    cout<<result;


    return 0;
}